internal/bus/bus_redis.go

```diff
@@ -290,18 +288,12 @@ func (r *redisReconcileSubscriptionsOp) run() {
 
 		r.mu.Lock()
 		if subscribeErr != nil {
-			for c := range subscribe {
-				r.dirtyChannels[c] = struct{}{}
-			}
+			maps.Copy(r.dirtyChannels, subscribe)
```

Comment:
not sure if both subscribe and unsubscribe err can happen. I am guessing copy does a copy and not merge? Is it possible for both errors to happen and `dirtyChannels` gets overwritten?

Reply:
it's a merge - https://cs.opensource.google/go/x/exp/+/9a3e6036:maps/maps.go;l=81 it's equivalent to the manual bit it replaced

Reply:
oh cool, :TIL: